### PR TITLE
feat(server): production end point

### PR DESCRIPTION
**Pull request by SiuYe**
#

## Changes this PR introduces

- Production end points

## How to test this PR

- Test all end points

## ClickUp reference

CU-nq08fz

## Notes for reviewers

- Thank you

## Checklist

- [x] Have I renamed the title of this pull request to follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)?
- [ ] I will squash and merge when my PR is approved.
- [x] I have added tests to cover my changes (if applicable).

### CODE REVIEW

**Comment by Mackbellemore** [comment]
Task linked: [CU-nq08fz Production page](https://app.clickup.com/t/nq08fz)

File: server/app/controllers/ProductionController.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+export class ProductionController extends BaseController {
+  constructor(@inject(TYPES.ProductionService) private productionService: ProductionService) {
+    super();
+  }
+
+  // @desc        Get all production
+  // @route       GET /productions
+  // @access      Public
+  @httpGet('/')
+  public async get(): Promise<results.JsonResult> {
```
Comment (gordonpn) [inline server/app/controllers/ProductionController.ts:28]: Can you add the `@Doc` like the other controllers have?

File: server/app/models/ProductionModel.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  option1: string;
+  option2: string;
+  quantity: string;
+  expectStartDate: Date;
+  expectedEndDate: Date;
+  actualStartDate: Date;
+  actualEndDate: Date;
+  manufacture: string;
+  note: string;
+}
```
Comment (gordonpn) [inline server/app/models/ProductionModel.ts:16]: Aren't a lot of these unneeded? @Mackbellemore 

Like manufacture? Aren't we the manufacturer? What's option1, option2 for? Component?

File: server/app/controllers/ProductionController.ts (hunk 1 above)
Comment (SiuYe) [reply server/app/controllers/ProductionController.ts:28]: of course, forgot

File: server/app/models/ProductionModel.ts (hunk 2 above)
Comment (SiuYe) [reply server/app/models/ProductionModel.ts:16]: The manufacturer would be which machine is producing the bike. For options, it's an additional feature that they would add such as a different paint color, or any option that ain't specified.

**Review by gordonpn — APPROVED** [review]

**Review by gordonpn — COMMENTED** [review]
Are you creating bikes/parts in this PR or in another one?

**Comment by gordonpn** [comment]
> Are you creating bikes/parts in this PR or in another one?

@SiuYe

**Review by KarinKazarian — COMMENTED** [review]
In the ticket, we have "Production is making bikes from parts, or parts from materials. Update the stock for parts and bikes." I don't see any checking or updating for Bikes or Parts, is this normal?

**Comment by SiuYe** [comment]
> Are you creating bikes/parts in this PR or in another one?

With this PR.

**Comment by SiuYe** [comment]
> In the ticket, we have "Production is making bikes from parts, or parts from materials. Update the stock for parts and bikes." I don't see any checking or updating for Bikes or Parts, is this normal?

Yes, because I didn't do it yet. It needs to be done...

**Pull request closed without merging**